### PR TITLE
fix: fix error when component === false

### LINKED ISSUE #45670: Passing `component={false}` to App show invalid prop warning
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/p/sandbox/recursing-hertz-lry38x?file=%2Fsrc%2FMainApp.tsx&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clojx2l5c000a3b6ieulkbg8o%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clojx2l5b00033b6i1tbc112b%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clojx2l5b00073b6ih16ieqtj%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clojx2l5b00093b6i8uw5n8ey%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clojx2l5b00033b6i1tbc112b%2522%253A%257B%2522id%2522%253A%2522clojx2l5b00033b6i1tbc112b%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FMainApp.tsx%2522%252C%2522id%2522%253A%2522clojx3yk1005x3b6i105hh7ws%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clojx3yk1005x3b6i105hh7ws%2522%257D%252C%2522clojx2l5b00093b6i8uw5n8ey%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clojx2l5b00083b6i8a8khd23%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clojx2l5b00093b6i8uw5n8ey%2522%252C%2522activeTabId%2522%253A%2522clojx2l5b00083b6i8a8khd23%2522%257D%252C%2522clojx2l5b00073b6ih16ieqtj%2522%253A%257B%2522id%2522%253A%2522clojx2l5b00073b6ih16ieqtj%2522%252C%2522activeTabId%2522%253A%2522clojx2l5b00043b6ilbv67bim%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clojx2l5b00043b6ilbv67bim%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%252C%257B%2522id%2522%253A%2522clojx2l5b00053b6i15p43rk2%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522build%2522%257D%252C%257B%2522id%2522%253A%2522clojx2l5b00063b6ivdra5a8h%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522preview%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522yarn%2520add%2520antd%25405.11.0%2522%252C%2522id%2522%253A%2522clojx32bz009z3b6ixipxsoaf%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A17.38095238095238%257D)

### Steps to reproduce

Wrap the app with `App (Application Manger)` and pass `component={false}`.

### What is expected?

App should not create dom node and no prop should be passed to `React.Fragment` if `component` is to `false`.

### What is actually happening?

Throw a warning:
```
Warning: Invalid prop `className` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.
```

| Environment | Info |
| --- | --- |
| antd | 5.11.0 |
| React | 18.2.0 |
| System | Mac M1 |
| Browser | Chrome |

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- close #45670

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | fix: fix error when component === false |
| 🇨🇳 Chinese |          - |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at f8cb9e6</samp>

This pull request improves the test coverage and robustness of the app component. It adds a new test case for when the `appConfig` prop is false, and modifies the `components/app/index.tsx` file to handle this scenario correctly.

### <samp>🤖 Generated by Copilot at f8cb9e6</samp>

*  Add and modify test cases for the app component in `components/app/__tests__/index.test.tsx` ([link](https://github.com/ant-design/ant-design/pull/45671/files?diff=unified&w=0#diff-52400589dcfd21bced4d2c624c0fbb235acac6ec308f1cf4c3fa37fcc8e80cf8R225), [link](https://github.com/ant-design/ant-design/pull/45671/files?diff=unified&w=0#diff-52400589dcfd21bced4d2c624c0fbb235acac6ec308f1cf4c3fa37fcc8e80cf8L230-R233), [link](https://github.com/ant-design/ant-design/pull/45671/files?diff=unified&w=0#diff-95b0dcfbdd9a195d1d0141272a065d901452a60a60632bd0e30f74e80eb85a6bL77-R77))
  - Mock console.error to suppress warnings ([link](https://github.com/ant-design/ant-design/pull/45671/files?diff=unified&w=0#diff-52400589dcfd21bced4d2c624c0fbb235acac6ec308f1cf4c3fa37fcc8e80cf8R225))
  - Assert that no warnings and no ant-app class are rendered when appConfig is false ([link](https://github.com/ant-design/ant-design/pull/45671/files?diff=unified&w=0#diff-52400589dcfd21bced4d2c624c0fbb235acac6ec308f1cf4c3fa37fcc8e80cf8L230-R233))
  - Restore console.error after the test ([link](https://github.com/ant-design/ant-design/pull/45671/files?diff=unified&w=0#diff-52400589dcfd21bced4d2c624c0fbb235acac6ec308f1cf4c3fa37fcc8e80cf8L230-R233))
  - Pass undefined as rootProps when component is false ([link](https://github.com/ant-design/ant-design/pull/45671/files?diff=unified&w=0#diff-95b0dcfbdd9a195d1d0141272a065d901452a60a60632bd0e30f74e80eb85a6bL77-R77))

### CODE REVIEW

File: components/app/index.tsx [hunk 1]
```diff
@@ -74,7 +74,7 @@ const App: React.FC<AppProps> & { useApp: () => useAppProps } = (props) => {
   return wrapSSR(
     <AppContext.Provider value={memoizedContextValue}>
       <AppConfigContext.Provider value={mergedAppConfig}>
-        <Component {...rootProps}>
+        <Component {...(component === false ? undefined : rootProps)}>
           {ModalContextHolder}
```
Comment (crazyair) [inline components/app/index.tsx:78]: 68行可以省略，在这里只需要一个三元表达式就可以

**Comment by crazyair** [comment]
奇怪，这个如果传了 false 有报错吗 https://github.com/react-component/field-form/blob/master/src/Form.tsx

File: components/app/index.tsx (hunk 1 above)
Comment (li-jia-nan) [reply components/app/index.tsx:78]: 是这个意思吗？

```tsx
 return wrapSSR(
    <AppContext.Provider value={memoizedContextValue}>
      <AppConfigContext.Provider value={mergedAppConfig}>
        {component === false ? (
          <>
            {ModalContextHolder}
            {messageContextHolder}
            {notificationContextHolder}
            {children}
          </>
        ) : (
          <Component {...rootProps}>
            {ModalContextHolder}
            {messageContextHolder}
            {notificationContextHolder}
            {children}
          </Component>
        )}
      </AppConfigContext.Provider>
    </AppContext.Provider>,
  );
```

**Comment by li-jia-nan** [comment]
> 奇怪，这个如果传了 false 有报错吗 https://github.com/react-component/field-form/blob/master/src/Form.tsx

没懂，这里传了 false 只是隐藏了dom，跟 rc-form 有啥关系吗？

**Comment by crazyair** [comment]
> > 奇怪，这个如果传了 false 有报错吗 https://github.com/react-component/field-form/blob/master/src/Form.tsx
> 
> 没懂，这里传了 false 只是隐藏了dom，跟 rc-form 有啥关系吗？

rc-form 看起来也有这个问题

**Comment by crazyair** [comment]
如果 rc-form 没有这个问题，那就是 React.Fragment 这个去掉，就是不用判断 false，应该就能解决这个 issue 的问题

**Review by kiner-tang — APPROVED** [review]

**Comment by li-jia-nan** [comment]
> 如果 rc-form 没有这个问题，那就是 React.Fragment 这个去掉，就是不用判断 false，应该就能解决这个 issue 的问题

这里有三个节点，React.Fragment 没办法去掉，就算不判断 false，也得加一个 <></>

**Comment by crazyair** [comment]
那就加

**Pull request merged**